core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -433,13 +390,19 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
 
   def checkCurrentWorkflowStatusCounts(implicit executionContext: ExecutionContext): Future[SaveCurrentWorkflowStatusCounts] = {
     datasource.inTransaction { dataAccess =>
-      dataAccess.workflowQuery.countWorkflowsForSubmissionByQueueStatus(submissionId).map { workflowStatusCounts =>
-        workflowStatusCounts.map { case (k, v) => WorkflowStatuses.withName(k) -> v }
+      for {
+        wfStatuses <- dataAccess.workflowQuery.countWorkflowsForSubmissionByQueueStatus(submissionId)
+        workspace <- getWorkspace(dataAccess).map(_.getOrElse(throw new RawlsException(s"workspace for submission $submissionId not found")))
+        subStatuses <- dataAccess.submissionQuery.countByStatus(SlickWorkspaceContext(workspace))
```

Comment:
Modified this method to return a `(Map[WorkflowStatus, Int], Map[SubmissionStatus, Int])`. Previously it was just counting workflow statuses.